__tests__/secret.test.js

```diff
@@ -89,4 +89,15 @@ describe('Tozny identity client', () => {
     expect(ops.createSecret(secretNameInvalid)).rejects.toThrow()
     expect(ops.createSecret(secretValueEmpty)).rejects.toThrow()
   })
+  it('can create a secret, and list it', async () => {
+    const secret = {
+      secretType: 'Credential',
+      secretName: `test-secret-${uuidv4()}`,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    await ops.createSecret(realmConfig, identity, secret)
+    await new Promise(r => setTimeout(r, 1000))
```

Comment:
hmm -- I admit I haven't paid as much attention on tests and should have. Why do we need this Timeout? I think in the search we have a method to have searches try until they find their results for a set amount of time or something (guessing it is here to allow search service to populate). This is a little less brittle than a timeout, which is not guaranteed by search.

Reply:
This was the solution @willmichael  and i could find, since a secret is added and immediately is searched. We needed the small gap to allow search to propagate

Reply:
https://github.com/tozny/js-sdk/blob/f0254f256f8dc996c0adb4d3e660df4d1ee1d8b0/__tests__/search.test.js#L28 - is where the loop is. This could get refactored into a search utility which runs the next in a loop until the results come back or it times out, and fail only if the timeout is hit.

It's basically an improvement on the straight timeout by allowing you to set more potential time for indexing to occur while still allowing things to move more quickly when indexing occurs quickly.

e.g.
```js
/**
 * Get the results from a query, using a loop to ensure they have time to index
 */
async function waitForNext(query) {
  // Start with a very short delay as immediate fetch of results right after writing
  // sometimes failes, but even with a very short window of wait it can succeed
  // first try.
  await new Promise(r => setTimeout(r, i * 100))
  // 10-second timeout period, 20 rounds of half a second between tries.
  for (let i = 0; i < 20; i++) {
    const found = await query.next()
    if (found.length > 0) {
      break
    }
    // delay half a second between tries
    await new Promise(r => setTimeout(r, i * 500))
  }
  return found
}
  ```